etl/definitions.py

```diff
@@ -1,10 +1,44 @@
-from dagster import Definitions, load_assets_from_modules
+from pathlib import Path
+
+from dagster import Definitions, EnvVar, load_assets_from_modules
+
+from etl.models.types.enrichment_type import EnrichmentType
+from etl.models.types.record_type import RecordType
 
 from . import assets
 from .jobs import embedding_job
+from .resources import (
+    InputDataFilesConfig,
+    OpenAiPipelineConfig,
+    OpenAiSettings,
+    OutputConfig,
+)
 
-all_assets = load_assets_from_modules([assets])
-all_jobs = [embedding_job]
-
+openai_settings = OpenAiSettings(openai_api_key=EnvVar("OPENAI_API_KEY").get_value(""))
 
-definitions = Definitions(assets=all_assets, jobs=all_jobs)
+definitions = Definitions(
+    assets=load_assets_from_modules([assets]),
+    jobs=[embedding_job],
+    resources={
+        "input_data_files_config": InputDataFilesConfig.from_env_vars(
+            data_files_directory_path_default=Path(__file__).parent.absolute()
+            / "data"
+            / "input"
+            / "data_files",
+            data_file_names_default=("mini-wikipedia.output.txt",),
+        ),
+        "openai_settings": openai_settings,
+        "openai_pipeline_config": OpenAiPipelineConfig(
+            openai_settings=openai_settings,
+            record_type=EnvVar("RECORD_TYPE").get_value(default=RecordType.WIKIPEDIA),
```

Comment:
Nit: it's good form to prefix environment variables with the name of your app, since it's a global namespace. Maybe `WIKIPEDIA_TRANSFORM_RECORD_TYPE`.